### DISCUSSION TITLE
mRNA data query

### DISCUSSION THREAD

**Discussion by jjgao** [cBioPortal/cbioportal#8216] [General]
Moving from issue https://github.com/cBioPortal/cbioportal/issues/5528 to Discussions.

This discussion was inspired by [a question from a user about inconsistency of mRNA data from different prostate studies](https://groups.google.com/d/msgid/cbioportal/c4fdbeb94812f59337870a4bb2607c88%40mail.irb.hr) and [another question](https://groups.google.com/forum/#!topic/cbioportal/kGNyVgOeutQ) Here is part of the discussion copied from emails.

JJ:
This type of question has come often. I think the term up/down-regulation is especially confusing as it indicates that the gene is dysregulated in the sample (similarn to mutations / CNAs). It's concerning when people just query genes by the default parameters and take the % alterations as a meaningful number. I am wondering how we can improve... Maybe making the threshold selection more interactive? Maybe add a note to the result?

Niki:
Good point.  I have two ideas.
1. Can we change up/down-regulation to high/low expression? Would that help? [fixed #5529]
2. I like you idea about making the threshold selection more interactive. Maybe we can disable the default threshold, and ask users to enter their own? We can have a little window that explain the process and associated problems. 

JJ:
I was thinking of # 1 too.

For # 2, it may not be that simple since each gene's threshold may need to be defined separately since different genes have different distributions. I am playing with the following ideas based # 2:

3. Use percentiles instead of z-score by default, e.g. use 25% and 75% by default (with options to change) - percentiles might be less confusing. (we could still support z-scores but it'd be not the default option.)
4. In OncoPrint, we could add some note to explain the process and potential issues. We then give users the option to change/adjust the threshold of each gene (submenu of the gene) -- we could make it interactive, e.g. display the distribution for a user to choose.
5. No matter if we do # 2-4, we could also consider creating a profile of expression outliers for querying.

Pichai:
I actually really like the idea of using percentiles, it makes less assumptions about the distribution of the gene and most people have an intuitive understanding of what percentiles are/mean. Wondering if you could also have a category for outliers (75th Percentile + IQR) by default, as I could see people being very interested in that.

cc'ing @cBioPortal/product

**Comment by jjgao** [comment]
tmazor commented on Dec 27, 2018
I think these are all great ideas - in particular highlighting the distribution and letting the user define the thresholds directly on the distribution seems helpful for putting the high/low calls in context.

**Comment by jjgao** [comment]
jjgao commented on Feb 18, 2019 

Some more thoughts.

- expression transformation options
  - percentiles (default)
  - original values
  - log2 values (only rnaseq?)
  - z-scores of original values
  - z-scores of log2 values
  - outlier based on original values
	- outlier is below 1.5 [IQR](https://www.wikiwand.com/en/Interquartile_range) of lower quartile or above 1.5 IQR of upper quartile
  - outlier based on log2 values
- reference population
  - A reference population is needed for all options above except original value and log2 values
  - The reference population should based on all samples in a profile, not just the selected ones. (should we precalculate since the values won't change?)
  - reference population options
    - all samples in the same expression profile
    - all samples in the same expression profile that is diploid of the gene in question (if discrete copy number data is available)
    - normal samples in the same study (if available)
- threshold
  - two threshold for high and low
![image](https://user-images.githubusercontent.com/840895/52978523-b4581980-339f-11e9-8eac-940a53cae75c.png)

We have two implementation options: 1) precalculating all transformation profiles or 2) calculating on the fly.
  - If precalculating all transformation profiles, many new files will be generated but there will not be many change to the website code.
  - If calucating on the fly, zscore files can be removed and a lot of changes need to be made in the code including API and query interface.
  - There may be other pros and cons.

This issue is also related to the [dynamic z-score calculation RFC](https://docs.google.com/document/d/1yNYDZhrC1qLrjogoS5M6BOW4A313D5u_Or6jhe5pNtc/edit) 

notifying @n1zea144

**Reply by pieterlukasse** [reply]
hi @jjgao, @schultzn, @tmazor, thanks for the overview! Here is my question for option 2: 
- would "calculating on the fly" unlock some of the mRNA expression analysis features that are currently disabled in the UI? E.g. mRNA heatmap is disabled and also one option in the comparison tab is currently disabled.
